data_service/src/types.rs

```diff
@@ -1,5 +1,25 @@
+use chrono::{NaiveDateTime, TimeDelta};
+
 #[derive(PartialEq)]
 pub struct PuuidToChampionMapping {
     pub puuid: String,
     pub champion_name: String,
+    pub position: String,
+}
+
+#[derive(PartialEq)]
+pub struct MatchTimelineFrame {
+    pub mapping: PuuidToChampionMapping,
+    /**
+     * Offset of game start time
+     */
+    pub frame_time: TimeDelta,
+    pub current_gold: i32,
+}
+
+#[derive(PartialEq)]
+pub struct MatchTimeline {
+    pub frames: Vec<MatchTimelineFrame>,
+    pub match_id: String,
+    pub start_time: NaiveDateTime,
 }
```

Comment:
The current API seemingly assumes that we monitor a singular champion for every frame in the timeline. is that correct? if yes, then which champion ( pass me the puuid ) ? if we want it tracked for every champion then either we create 10 timelines ( for all champions ) or we have a vector of champions for every frame. Please clarify this. it's a bit confusing